client/src/components/ContentNavigator/convert.ts

```diff
@@ -0,0 +1,349 @@
+        codeOptions: {
+          ...baseNode.arguments.codeOptions,
+          code: entry.code,
+        },
+      },
+    };
+
+    flowData.nodes[idNode] = stepNode;
+    arrayIdNode.push(idNode);
+
+    if (idx > 0) {
+      const connection = {
+        sourcePort: {
+          node: arrayIdNode[idx - 1],
+          portName: "outTables",
+          index: 0,
+        },
+        targetPort: {
+          node: arrayIdNode[idx],
+          portName: "inTables",
+          index: 0,
+        },
+      };
+      flowData.connections.push(connection);
+    }
+  }
+  return flowData;
+}
+
+function generateFlowDataSwimlane(inputList: Entry[], outputFile: string) {
+  const now = new Date();
+  const nowString = now.toISOString();
+
+  const flowData = { ...baseFlow };
+  flowData.creationTimeStamp = nowString;
+  flowData.modifiedTimeStamp = nowString;
+  flowData.name = outputFile;
+  flowData.properties = {
+    ...baseFlow.properties,
+    UI_PROP_DF_ID: "58e4d421-705d-448c-b397-cc8c9fab6c48",
+    UI_PROP_DF_EXECUTION_ORDERED: "true",
+  };
+  flowData.nodes = {};
+  flowData.connections = [];
+
+  for (let idx = 0; idx < inputList.length; idx++) {
+    const swimlaneName = "Notebook Cell " + (idx + 1).toString();
+    const idNode = v4();
+    const swimlane = {
+      ...baseSwimlane,
+      id: idNode,
+      name: swimlaneName,
+      priority: idx + 1,
+      dataFlowAndBindings: {
+        ...baseSwimlane.dataFlowAndBindings,
+        dataFlow: generateFlowDataNode([inputList[idx]], null),
+      },
+    };
+    flowData.nodes[idNode] = swimlane;
+  }
+  return flowData;
+}
+
+function generateFlowData(inputList: Entry[], outputFile: string) {
+  const flowConversionMode = workspace
+    .getConfiguration("SAS")
+    .get("flowConversionMode");
+  if (flowConversionMode === "Node") {
+    return generateFlowDataNode(inputList, outputFile);
+  }
+  return generateFlowDataSwimlane(inputList, outputFile);
+}
+
+function generateCodeListFromSASNotebook(content: string): Entry[] {
+  const codeList = [];
+  try {
+    const notebookContent = JSON.parse(content);
+
+    for (const cell of notebookContent) {
+      let code = cell.value;
+      if (code !== "") {
+        const language = cell.language;
+        if (["python", "sas", "sql"].includes(language)) {
+          if (language === "sql") {
+            code = `PROC SQL;
+${code};
+QUIT;`;
+          }
+          codeList.push({ code, language });
+        }
+      }
+    }
+  } catch (error) {
+    console.error("Error reading or parsing the .sasnb file:", error);
+  }
+  return codeList;
+}
+
+function generateCodeListFromPythonNotebook(content: string): Entry[] {
+  const codeList = [];
+  try {
+    const notebookContent = JSON.parse(content);
+
+    for (const cell of notebookContent.cells) {
+      const code = cell.source.join("");
+      const cellType = cell.cell_type;
+      if (code !== "" && cellType === "code") {
+        const language = "python";
```

Comment:
We have [sas_kernel](https://github.com/sassoftware/sas_kernel) for Jupyter. Is it possible to have sas here?

Reply:
done in 5d9b2307a23ae99f5bed029eb98f4a11face2a07